### ISSUE TITLE
Can't retrieve a time slice with multi ensemble member Data Array 

### ISSUE THREAD

**Issue by EricKeenan** [NCAR/intake-esm#159]
FYI this issue occurs on casper (DAV).

I can succesfully time slice with a single ensemble member: 

```
# Define time domain
t0 = '2009-01-01'
tf = '2009-12-31'

# Retrieve single ensemble member
cat = intake.open_esm_datastore("../catalogs/glade-cmip6.json")
cat = cat.search(experiment_id = ['historical'], source_id = ['CESM2'], \
                  table_id = ['day'], variable_id = ['pr'], member_id = ['r2i1p1f1'])

train_ds = cat.to_dataset_dict(cdf_kwargs={'chunks': {'time': 1000}})
train_ds = train_ds['CMIP.NCAR.CESM2.historical.day.gn']['pr']
train_ds = train_ds.sel(time=slice(t0, tf))
```

However, when I use multiple ensemble members I get the following error: 

```
# Define time domain
t0 = '2009-01-01'
tf = '2009-12-31'

# Retrieve single ensemble member
cat = intake.open_esm_datastore("../catalogs/glade-cmip6.json")
cat = cat.search(experiment_id = ['historical'], source_id = ['CESM2'], \
                  table_id = ['day'], variable_id = ['pr'])

train_ds = cat.to_dataset_dict(cdf_kwargs={'chunks': {'time': 1000}})
train_ds = train_ds['CMIP.NCAR.CESM2.historical.day.gn']['pr']
train_ds = train_ds.sel(time=slice(t0, tf))
```
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/miniconda3/envs/xsd/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_slice_bound(self, label, side, kind)
   5166             try:
-> 5167                 return self._searchsorted_monotonic(label, side)
   5168             except ValueError:

~/miniconda3/envs/xsd/lib/python3.7/site-packages/pandas/core/indexes/base.py in _searchsorted_monotonic(self, label, side)
   5127 
-> 5128         raise ValueError("index must be monotonic increasing or decreasing")
   5129 

ValueError: index must be monotonic increasing or decreasing

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)
<ipython-input-52-82a85fbde6ae> in <module>
----> 1 train_ds = train_ds.sel(time=slice(t0, tf))

~/miniconda3/envs/xsd/lib/python3.7/site-packages/xarray/core/dataarray.py in sel(self, indexers, method, tolerance, drop, **indexers_kwargs)
   1045             method=method,
   1046             tolerance=tolerance,
-> 1047             **indexers_kwargs
   1048         )
   1049         return self._from_temp_dataset(ds)

~/miniconda3/envs/xsd/lib/python3.7/site-packages/xarray/core/dataset.py in sel(self, indexers, method, tolerance, drop, **indexers_kwargs)
   1998         indexers = either_dict_or_kwargs(indexers, indexers_kwargs, "sel")
   1999         pos_indexers, new_indexes = remap_label_indexers(
-> 2000             self, indexers=indexers, method=method, tolerance=tolerance
   2001         )
   2002         result = self.isel(indexers=pos_indexers, drop=drop)

~/miniconda3/envs/xsd/lib/python3.7/site-packages/xarray/core/coordinates.py in remap_label_indexers(obj, indexers, method, tolerance, **indexers_kwargs)
    390 
    391     pos_indexers, new_indexes = indexing.remap_label_indexers(
--> 392         obj, v_indexers, method=method, tolerance=tolerance
    393     )
    394     # attach indexer's coordinate to pos_indexers

~/miniconda3/envs/xsd/lib/python3.7/site-packages/xarray/core/indexing.py in remap_label_indexers(data_obj, indexers, method, tolerance)
    259             coords_dtype = data_obj.coords[dim].dtype
    260             label = maybe_cast_to_coords_dtype(label, coords_dtype)
--> 261             idxr, new_idx = convert_label_indexer(index, label, dim, method, tolerance)
    262             pos_indexers[dim] = idxr
    263             if new_idx is not None:

~/miniconda3/envs/xsd/lib/python3.7/site-packages/xarray/core/indexing.py in convert_label_indexer(index, label, index_name, method, tolerance)
    123             _sanitize_slice_element(label.start),
    124             _sanitize_slice_element(label.stop),
--> 125             _sanitize_slice_element(label.step),
    126         )
    127         if not isinstance(indexer, slice):

~/miniconda3/envs/xsd/lib/python3.7/site-packages/pandas/core/indexes/base.py in slice_indexer(self, start, end, step, kind)
   5032         slice(1, 3)
   5033         """
-> 5034         start_slice, end_slice = self.slice_locs(start, end, step=step, kind=kind)
   5035 
   5036         # return a slice

~/miniconda3/envs/xsd/lib/python3.7/site-packages/pandas/core/indexes/base.py in slice_locs(self, start, end, step, kind)
   5252         end_slice = None
   5253         if end is not None:
-> 5254             end_slice = self.get_slice_bound(end, "right", kind)
   5255         if end_slice is None:
   5256             end_slice = len(self)

~/miniconda3/envs/xsd/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_slice_bound(self, label, side, kind)
   5168             except ValueError:
   5169                 # raise the original KeyError
-> 5170                 raise err
   5171 
   5172         if isinstance(slc, np.ndarray):

~/miniconda3/envs/xsd/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_slice_bound(self, label, side, kind)
   5162         # we need to look up the label
   5163         try:
-> 5164             slc = self.get_loc(label)
   5165         except KeyError as err:
   5166             try:

~/miniconda3/envs/xsd/lib/python3.7/site-packages/xarray/coding/cftimeindex.py in get_loc(self, key, method, tolerance)
    330             return self._get_string_slice(key)
    331         else:
--> 332             return pd.Index.get_loc(self, key, method=method, tolerance=tolerance)
    333 
    334     def _maybe_cast_slice_bound(self, label, side, kind):

~/miniconda3/envs/xsd/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   2897                 return self._engine.get_loc(key)
   2898             except KeyError:
-> 2899                 return self._engine.get_loc(self._maybe_cast_indexer(key))
   2900         indexer = self.get_indexer([key], method=method, tolerance=tolerance)
   2901         if indexer.ndim > 1 or indexer.size > 1:

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine._get_loc_duplicates()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine._maybe_get_bool_indexer()

KeyError: cftime.DatetimeNoLeap(2009, 12, 31, 23, 59, 59, 999999, 1, 1)
```

**Comment by andersy005** [comment]
@EricKeenan, this issue seems to be fixed in #171.

Here's a notebook with details:
https://gist.github.com/andersy005/502bfd2b793baa1649fda28332c923f1#file-intake-esm-issue-159-ipynb

Once #171 is merged, feel free to update your environment by installing intake-esm from master branch

```bash
python -m pip install git+https://github.com/NCAR/intake-esm.git
```

**Comment by andersy005** [comment]
Addressed over in #171
